spec/unit/puppet-strings/markdown_spec.rb

```diff
@@ -2,56 +2,55 @@
 
 require 'spec_helper'
 require 'puppet-strings/markdown'
-require 'puppet-strings/markdown/table_of_contents'
 require 'tempfile'
 
 describe PuppetStrings::Markdown do
-  let(:fixture_path) do
-    File.expand_path("../../fixtures", __dir__)
-  end
+  describe 'rendering fixtures' do
+    let(:fixture_path) do
+      File.expand_path("../../fixtures", __dir__)
+    end
 
-  def fixture_content(fixture)
-    @fixtures ||= {}
-    @fixtures[fixture] ||= File.read(File.join(fixture_path, fixture))
-  end
+    def fixture_content(fixture)
+      @fixtures ||= {}
+      @fixtures[fixture] ||= File.read(File.join(fixture_path, fixture))
+    end
 
-  def parse_shared_content
-    # Populate the YARD registry with both Puppet and Ruby source
-    YARD::Parser::SourceParser.parse_string(fixture_content("puppet/class.pp"), :puppet)
-    YARD::Parser::SourceParser.parse_string(fixture_content("puppet/function.pp"), :puppet)
-    YARD::Parser::SourceParser.parse_string(fixture_content("ruby/func4x.rb"), :ruby)
-    YARD::Parser::SourceParser.parse_string(fixture_content("ruby/func4x_1.rb"), :ruby)
-    YARD::Parser::SourceParser.parse_string(fixture_content("ruby/func3x.rb"), :ruby)
-    YARD::Parser::SourceParser.parse_string(fixture_content("ruby/func3x.rb"), :ruby)
-    YARD::Parser::SourceParser.parse_string(fixture_content("ruby/provider.rb"), :ruby)
-    YARD::Parser::SourceParser.parse_string(fixture_content("ruby/resource_type.rb"), :ruby)
-    YARD::Parser::SourceParser.parse_string(fixture_content("ruby/resource_api.rb"), :ruby)
-
-    # task metadata derives the task name from the filename, so we have to parse
-    # directly from the filesystem to correctly pick up the name
-    YARD::Parser::SourceParser.parse(File.join(fixture_path, "json/backup.json"))
-  end
+    def parse_shared_content
+      # Populate the YARD registry with both Puppet and Ruby source
+      YARD::Parser::SourceParser.parse_string(fixture_content("puppet/class.pp"), :puppet)
```

Comment:
Isn't it tempting to rewrite this to a loop?

Reply:
Ha! This has already started snowballing a bit… I keep coming back and finding little tweaks, and then those bring more things in scope that I could tweak…